src/compositions/usePagination.ts

```diff
@@ -84,18 +84,22 @@ export function usePagination<
   const countSubscription = useSubscriptionWithDependencies(countMethod, countSubscriptionParameters, options)
   const total = computed(() => countSubscription.response ?? 0)
 
-  const fetchSubscriptionParameters = ref(getFetchSubscriptionParameters()) as Ref<Parameters<TFetch>[] | null>
+  const fetchParametersRef = toRef(fetchParametersGetter)
   const fetchSubscriptions: UseSubscription<TFetch>[] = reactive([])
   const results = computed(() => fetchSubscriptions.flatMap(subscription => subscription.response ?? []) as unknown as Awaited<ReturnType<TFetch>>)
 
-  watch([total, fetchSubscriptionParameters], ([total, parameters]) => {
-    if (parameters === null || total === 0) {
+  watch([total, page, fetchParametersRef], ([total, page, parameters]) => {
+    if (total === 0 || page === 0 || parameters === null) {
```

Comment:
I'm following the other two, but why `page === 0`? when would that occur?

Reply:
When in infinite mode the page starts at 0 and waits for the implementation to call `next`.

Reply:
ahhhh ok that makes sense